File: backend/app/services/conversation_service.py
from sqlalchemy.orm import Session

from app.database.models import Conversation


def create_conversation(
    db: Session,
    user_id: int,
    title: str = "New Chat",
):

    conversation = Conversation(
        title=title,
        user_id=user_id,
    )

    db.add(conversation)
    db.commit()
    db.refresh(conversation)

    return conversation


def get_conversations(
    db: Session,
    user_id: int,
):

    return (
        db.query(Conversation)
        .filter(
            Conversation.user_id == user_id
        )
        .order_by(
            Conversation.updated_at.desc()
        )
        .all()
    )


def get_conversation(
    db: Session,
    conversation_id: int,
    user_id: int,
):

    return (
        db.query(Conversation)
        .filter(
            Conversation.id == conversation_id,
            Conversation.user_id == user_id,
        )
        .first()
    )


def delete_conversation(
    db: Session,
    conversation: Conversation,
):

    db.delete(conversation)
    db.commit()

File: frontend/src/services/conversationService.js
import api from "./api";

export async function createConversation() {
  const response = await api.post("/conversations");
  return response.data;
}

export async function getConversations() {
  const response = await api.get("/conversations");
  return response.data;
}

export async function getConversationMessages(id) {
  const response = await api.get(`/conversations/${id}`);
  return response.data;
}

export async function deleteConversation(id) {
  const response = await api.delete(`/conversations/${id}`);
  return response.data;
}